File: Pathfinder.Core/Entities/Category.cs
using Pathfinder.Core.Entities.Base;
using System.ComponentModel.DataAnnotations;

namespace Pathfinder.Core.Entities
{
    public class Category : Entity
    {
        [Required, StringLength(80)]
        public string Name { get; set; }
        public string Description { get; set; }
    }
}

File: Pathfinder.Web/ClientApp/src/shared/models/Category.ts
import { ICategory } from './Interfaces/ICategory';
export class Category implements ICategory {
  public id: number;
  public name: string;
  public description: string;
  constructor(id: number, name: string, description: string) {
    this.id = id;
    this.name = name;
    this.description = description;
  }
}
